contracts/BatchExchangeViewer.sol

```diff
@@ -348,6 +349,71 @@ contract BatchExchangeViewer {
         return updateSellTokenBalance(element, sellTokenBalance);
     }
 
+    /**
+     * @dev Queries the token information for the given tokenId. Can handle symbols of type string and bytes (e.g. MKR).
+     * Returns "Unknown" as symbol value if it cannot be retrieved and reverts if decimals can not be fetched
+     * (to avoid ambiguity with a "valid" value).
+     * @param tokenId the ID of a token listed on BatchExchange
+     * @return the address, symbol and decimals of the token contract
+     */
+    function getTokenInfo(uint16 tokenId)
+        public
+        view
+        returns (
+            address,
+            string memory symbol,
+            uint8 decimals
+        )
+    {
+        address tokenAddress = batchExchange.tokenIdToAddressMap(tokenId);
+        symbol = "Unknown";
+        (bool hasStringSymbol, ) = address(this).staticcall(
+            abi.encodeWithSignature("getTokenSymbolString(address)", tokenAddress)
+        );
+        if (hasStringSymbol) {
+            symbol = getTokenSymbolString(tokenAddress);
+        } else {
+            (bool hasBytesSymbol, ) = address(this).staticcall(
+                abi.encodeWithSignature("getTokenSymbolBytes(address)", tokenAddress)
+            );
+            if (hasBytesSymbol) {
+                symbol = getTokenSymbolBytes(tokenAddress);
+            }
+        }
+        return (tokenAddress, symbol, ERC20Detailed(tokenAddress).decimals());
```

Comment:
I don't think defaulting to 18 is a good idea. If we e.g. in the price estimator start converting user input from base units into atoms with a wrong decimal we are likely going to get into trouble. I don't think there is a safe default value. Therefore, I'd rather be explicit that this token cannot be supported and revert. Do we know of tokens that don't implement decimals?

Reply:
> If we e.g. in the price estimator start converting user input from base units into atoms with a wrong decimal we are likely going to get into trouble. 

Hmm, good point. We could return the decimals in the result as well, so that the number of atoms can be re-computed, but that is probably a disaster waiting to happen...

> I'd rather be explicit that this token cannot be supported and revert.

I'm convinced as well now.